focus-android/app/src/main/java/org/mozilla/focus/session/PrivateNotificationFeature.kt

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.session
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.selector.privateTabs
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import kotlin.reflect.KClass
+
+/**
+ * Responsible for starting or stopping up a [SessionNotificationService]
+ * depending on whether a private tab is opened.
+ *
+ * @param browserStore Browser store reference used to observe the number of private tabs.
+ * @param sessionNotificationServiceClass The service sub-class that creates and controls the notification.
+ */
+class PrivateNotificationFeature(
+    context: Context,
+    private val browserStore: BrowserStore,
+    private val sessionNotificationServiceClass: KClass<SessionNotificationService>,
+    private val permissionRequestHandler: (() -> Unit),
+) : LifecycleAwareFeature {
+
+    private val applicationContext = context.applicationContext
+    private var scope: CoroutineScope? = null
```

Comment:
What's this used for? Seems it's only cancelled in `onStop()` 🤔

Reply:
Good question! I updated the code.